chain/chain/src/store.rs

```diff
@@ -74,6 +74,16 @@ pub enum GCMode {
     StateSync { clear_block_info: bool },
 }
 
+impl fmt::Debug for GCMode {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            GCMode::Fork(_) => write!(f, "GCMode::Fork"),
+            GCMode::Canonical(_) => write!(f, "GCMode::Canonical"),
+            GCMode::StateSync { .. } => write!(f, "GCMode::StateSync"),
```

Comment:
nit: You can use the f.debug_struct method. Under the hood it'll also use write but you're guaranteed that its style will be consistent with the rest of the codebase and the defaults. You can find plenty of examples in the codebase.